includes/class-patches.php

```diff
@@ -25,6 +28,120 @@ public static function init() {
 
 		// Disable WooCommerce image regeneration to prevent regenerating thousands of images.
 		add_filter( 'woocommerce_background_image_regeneration', '__return_false' );
+
+		// Jetpack Modules AMP Plus.
+		add_action( 'wp_enqueue_scripts', [ __CLASS__, 'jetpack_modules_amp_plus' ] );
+		add_action( 'wp_footer', [ __CLASS__, 'inject_jetpack_search_options_amp_plus' ] );
+		add_action( 'wp_footer', [ __CLASS__, 'jetpack_search_maybe_render_sort_js' ] );
+	}
+
+	/**
+	 * Enable AMP Plus for Jetpack Modules.
+	 */
+	public static function jetpack_modules_amp_plus() {
+		if ( ! AMP_Enhancements::should_use_amp_plus() ) {
+			return false;
+		}
+		$modules_scripts_handles = [
+			'jetpack-instant-search', // Jetpack Instant Search.
+			'jp-tracks',              // Tracks analytics library.
+			'wp-i18n',                // Jetpack Instant Search dependency.
+			'jetpack-search-widget',  // Jetpack Search widget.
+		];
+		add_filter(
+			'script_loader_tag',
+			function( $tag, $handle, $src ) use ( $modules_scripts_handles ) {
+				if ( in_array( $handle, $modules_scripts_handles, true ) ) {
+					return '<script data-amp-plus-allowed async src="' . $src . '"></script>';
+				}
+				return $tag;
+			},
+			10,
+			3
+		);
+	}
+
+	/**
+	 * Reinject Jetpack Instant Search options with AMP Plus enabled.
+	 *
+	 * Jetpack uses `wp_add_inline_script()`, which does not allow for custom tag
+	 * attributes, so we have to inject it ourselves.
```

Comment:
All frontend JS scripts pass via [`amp_validation_error_sanitized`](https://github.com/Automattic/newspack-plugin/blob/3a8cfa794806ee7d7b61ab4481587f51e631f60f/includes/class-amp-enhancements.php#L74) – this problem should be approached from there, as a [script-filter](https://github.com/Automattic/newspack-plugin/blob/3a8cfa794806ee7d7b61ab4481587f51e631f60f/includes/class-amp-enhancements.php#L81) (instead of enqueueing the scripts explicitly). The scripts used by this Jetpack's feature seem to have `id`s prefixed with `jetpack-instant-search-` and `wp-jp-i18n-`, e.g.:

<img width="1414" alt="image" src="https://user-images.githubusercontent.com/7383192/152758011-746167ca-c9e2-4341-b896-440fda888ca0.png">

(this is the value of the `$error` argument to `amp_validation_error_sanitized` filter)

Reply:
Good call! I didn't know that was how AMP Plus was handled! 325b0a3367b5559a3f444fc35cd8c2d5605ba583 adds a filter hook to the sanitization handler and parses the jetpack errors using IDs and texts. Some of the inline scripts do not have ID so it's parsing the script text to match.

Reply:
I also removed any of the i18n scripts from this PR because `wp-` scripts are already handled by default.